src/test/java/org/example/service/ApplicationServiceTest.java

```diff
@@ -0,0 +1,134 @@
+package org.example.service;
+
+import org.example.daos.ApplicationDao;
+import org.example.exceptions.DoesNotExistException;
+import org.example.exceptions.FailedToCreateException;
+import org.example.exceptions.InvalidException;
+import org.example.models.ApplicationRequest;
+import org.example.models.ApplicationResponse;
+import org.example.services.ApplicationService;
+import org.example.validators.ApplicationValidator;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+
+import java.sql.SQLException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.doNothing;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.doThrow;
+
+public class ApplicationServiceTest {
+
+    ApplicationDao applicationDao = Mockito.mock(ApplicationDao.class);
+    ApplicationValidator applicationValidator = Mockito.mock(ApplicationValidator.class);
+
+    ApplicationService applicationService = new ApplicationService(applicationDao, applicationValidator);
+
+    ApplicationRequest applicationRequest = new ApplicationRequest(
+            "user@kainos.com",
+            1,
+            "https://academy-job-portal-cvs.s3.eu-west-1.amazonaws.com/the_extra_semicolon/TestDocument.docx"
```

Comment:
Look at previous comments, remove URL as you are exposing it to the world. Why? :(